libs/report/committers/src/lib/parse-monthly.ts

```diff
@@ -1,44 +1,29 @@
-import {
-  eachMonthOfInterval,
-  format,
-  isWithinInterval,
-  max,
-  min,
-} from 'date-fns';
-import { MonthlyData } from './types';
-
-export function parseMonthly(
-  startDate: Date,
-  endDate: Date,
-  entries: { commitHash: string; committer: string; date: string }[]
-) {
+import { eachMonthOfInterval, format, isWithinInterval } from 'date-fns';
+import { Commit, MonthlyData } from './types';
+
+export function parseMonthly(startDate: Date, endDate: Date, entries: Commit[]) {
   const monthly: MonthlyData[] = [];
-  const dates = [startDate, endDate];
-  const ival = {
-    start: min(dates),
-    end: max(dates),
-  };
-  const range = eachMonthOfInterval(ival);
-  for (const idxr in range) {
-    const idx = parseInt(idxr);
-    if (idx + 1 >= range.length) {
-      continue;
-    }
-    const [start, end] = [range[idx], range[idx + 1]];
+  const range = eachMonthOfInterval({
+    start: startDate,
+    end: endDate,
+  });
+
+  for (let idx = 0; idx < range.length - 1; idx++) {
+    // `convertToUTC` is an awful hack to convert the date to UTC
+    // This is needed because the date-fns eachMonthOfInterval function
+    // is not timezone aware and will return the date in the local timezone
+    // Refs: https://github.com/date-fns/date-fns/issues/3097
+    const [start, end] = [convertToUTC(range[idx]), convertToUTC(range[idx + 1])];
```

Comment:
I don't think it is right to do this `convertToUTC()` transformation, unless we also do it to the initial before-/after-dates that we pass to the git command.

Everywere else, we accept that we are dealing with dates that represent start/end of periods in the user's timezone. We should either stick to that or convert the dates from the beginning to represent the start/end of periods in UTC.

Reply:
There is no timezone where the end of the month start at `11pm` thats the issue 😄 so I don't think its acceptable